File: cplex/examples/src/java/MIPex2.java
/* --------------------------------------------------------------------------
 * File: MIPex2.java
 * Version 12.6.1  
 * --------------------------------------------------------------------------
 * Licensed Materials - Property of IBM
 * 5725-A06 5725-A29 5724-Y48 5724-Y49 5724-Y54 5724-Y55 5655-Y21
 * Copyright IBM Corporation 2001, 2014. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or
 * disclosure restricted by GSA ADP Schedule Contract with
 * IBM Corp.
 * --------------------------------------------------------------------------
 *
 * MIPex2.java - Reading in and optimizing a MIP problem.  In fact, this
 *               program also works for LP or QP problems, but is different
 *               from LPex2 in that no dual solution information is queried.
 *
 * To run this example, command line arguments are required.
 * i.e.,   java MIPex2  filename
 * where 
 *     filename is the name of the file, with .mps, .lp, or .sav extension
 * Example:
 *     java MIPex2  mexample.mps
 */

import ilog.concert.*;
import ilog.cplex.*;


public class MIPex2 {
   static void usage() {
      System.out.println("usage:  MIPex2 <filename>");
   }

   public static void main(String[] args) {
      if ( args.length != 1 ) {
         usage();
         return;
      }
      try {
         IloCplex cplex = new IloCplex();
       
         cplex.importModel(args[0]);
       
         if ( cplex.solve() ) {
            System.out.println("Solution status = " + cplex.getStatus());
            System.out.println("Solution value  = " + cplex.getObjValue());
          
            // Access the IloLPMatrix object that has been read from a file in
            // order to access variables which are the columns of the LP.  The
            // method importModel() guarantees that exactly one IloLPMatrix
            // object will exist, which is why no tests or iterators are
            // needed in the following line of code.

            IloLPMatrix lp = (IloLPMatrix)cplex.LPMatrixIterator().next();
          
            double[] x = cplex.getValues(lp);
            for (int j = 0; j < x.length; ++j) {
               System.out.println("Variable " + j + ": Value = " + x[j]);
            }
         }
         cplex.end();
      }
      catch (IloException e) {
         System.err.println("Concert exception caught: " + e);
      }
   }
}

File: cplex/examples/src/matlab/mipex2.m
function mipex2(varargin)
% Read and optimize a mixed integer programming problem
%
% To run this example, a function argument is required.
%    i.e.,   mipex2(filename)
%    where
%       filename is the name of the problem file, with .mps, .lp, or .sav
%       extension
%    Example:
%       mipex2(mexample.mps)

% ---------------------------------------------------------------------------
% File: mipex2.m
% Version 12.6.1
% ---------------------------------------------------------------------------
% Licensed Materials - Property of IBM
% 5725-A06 5725-A29 5724-Y48 5724-Y49 5724-Y54 5724-Y55 5655-Y21
% Copyright IBM Corporation 2008, 2014. All Rights Reserved.
%
% US Government Users Restricted Rights - Use, duplication or
% disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
% ---------------------------------------------------------------------------

try
   % Check the command line arguments
   if nargin ~= 1
      display ('Usage: mipex2(filename)');
      display ('where filename is a file with extension: ');
      display ('MPS, SAV, or LP (lower case is allowed)');
      display ('Exiting...')
      return;
   else
      m = varargin{1};
   end
   
   % Initialize the CPLEX object
   cpx = Cplex('mipex2');
   
   % Now read the file and copy the data into cpx.Model
   cpx.readModel(m);
   
   % Optimize the problem
   cpx.solve();
   
   % Write the solution
   fprintf ('\nSolution status = %s \n', cpx.Solution.statusstring);
   fprintf ('Solution value = %f \n', cpx.Solution.objval);
   disp ('Values =');
   disp (cpx.Solution.x);
catch m
   throw (m);
end
end
